### PR TITLE
Bundling warnings

### LINKED ISSUE #1102: vTHL3: Adding and removing resources to submitted application
(split out from #842)

- [x] when the application is being processed, the handler can add and remove resources
- [ ] only the handler can add resources where the form is different to the one in the application
- [ ] if the new resources use a different form, give a warning

### PR DESCRIPTION
**Pull request by Macroz**
Implements bundling warnings from #1102 for both applicant and handler.

Closes #1102

### CODE REVIEW

**Comment by hukka** [comment]
A bit oddly, this will give the bundling warning when no resources are selected

**Comment by hukka** [comment]
Could we somehow show the user which resources are ok to add? Now the software is just yelling at the user for not knowing beforehand.

File: src/clj/rems/workflow/dynamic.clj [hunk 1]
```diff
@@ -78,10 +78,22 @@
...
+  "Checks the given catalogue items for validity and merges the errors"
   [catalogue-item-ids injections]
   (apply merge-with into (keep #(invalid-catalogue-item-error % injections) catalogue-item-ids)))
 
+(defn- unbundlable-catalogue-items
+  "Checks that the given catalogue items are bundlable by the given actor."
+  [application catalogue-item-ids actor injections]
+  (let [catalogue-items (map (:get-catalogue-item injections) catalogue-item-ids)
+        handlers (get-in application [:application/workflow :workflow.dynamic/handlers])]
+    (prn :kakka handlers actor)
```
Comment (hukka) [inline src/clj/rems/workflow/dynamic.clj]: Dev debug print?

File: test/clj/rems/workflow/test_dynamic.clj [hunk 2]
```diff
@@ -190,6 +191,13 @@
                            {:type :application.command/change-resources
                             :actor applicant-user-id
                             :catalogue-item-ids [1 2]}
+                           injections)))
+      (is (= {:errors [{:type :unbundlable-catalogue-items
+                        :catalogue-item-ids [1 2 3]}]}
+             (fail-command application
+                           {:type :application.command/change-resources
+                            :actor applicant-user-id
+                            :catalogue-item-ids [1 2 3]}
```
Comment (hukka) [inline test/clj/rems/workflow/test_dynamic.clj]: Would it make sense to have separate tests for different form and different workflow? Or is that overkill?

**Comment by hukka** [comment]
While the handler can bundle unbundlable resources, the message in the yellow box is still "The resources can't be bundled into one application because they have different forms or workflows." which implies that it is not possible. Maybe just "Did you mean to bundle these resources? They don't have the same workflow or form" or something like that.

**Comment by Macroz** [comment]
Ready for review again.

**Pull request merged**